09_json/willshen8/pkg/puppy/json/jsonifyPuppy.go

```diff
@@ -0,0 +1,22 @@
+package puppy
+
+import (
+	"encoding/json"
+
+	puppy "github.com/anz-bank/go-course/09_json/willshen8/pkg/puppy"
+	store "github.com/anz-bank/go-course/09_json/willshen8/pkg/puppy/store"
+)
+
+func UnmarshalPuppy(jsonData []byte, decodedPuppies *[]puppy.Puppy) error {
+	if unMarshalError := json.Unmarshal(jsonData, &decodedPuppies); unMarshalError != nil {
+		return unMarshalError
+	}
+	return nil
+}
+
+func SaveJSONPuppyToStore(p *puppy.Puppy, m *store.MapStore) error {
```

Comment:
How is that puppy a JSON puppy? 
same as above: `return m.CreatePuppy(p)` is nearly all you need (minus the ignored ID) so you might as well just use it.

Reply:
Corrected, thanks.